### PR TITLE
fix: TabBar issues

### PR DESCRIPTION
**Pull request by kazo0**
closes https://github.com/unoplatform/uno.toolkit.ui/issues/501

## PR Type

What kind of change does this PR introduce?
- Bugfix

## Changes

- Fix for calling ContainerFromIndex with a -1 value
- Fix for TabBarSelectorBehavior interfering with sliding indicator

### CODE REVIEW

**Review by agneszitte-nventive — APPROVED** [review]
Some tests can be added for these changes ?

**Comment by kazo0** [comment]
> Some tests can be added for these changes ?

I'm not sure how to test the sliding indicator, need to think about it

**Comment by Xiaoy312** [comment]
set selection index, and `WaitFor` the animated value (eg: transform.x) to reach the expected value?

**Comment by kazo0** [comment]
yeah thats what i am doing now, i was thinking more about the "in between" sliding when you are scrolling a FlipView and it should slide along based on the scroll offset. Testing for the indicator moving from item to item on selection is fine but the issue here was that it wasn't properly sliding to its destination

File: src/Uno.Toolkit.UI/Controls/TabBar/TabBarSelectionIndicatorPresenter.cs [hunk 1]
```diff
@@ -343,6 +349,16 @@ private void UpdateSelectionIndicatorPosition(TabBarItem? destination = null)
...
 		}
 
+		private Point? GetRelativePosition(UIElement? element)
+		{
+			if (element == null)
+			{
+				return null;
+			}
+
+			return element.TransformToVisual(Owner).TransformPoint(default);
```
Comment (jeromelaban) [inline src/Uno.Toolkit.UI/Controls/TabBar/TabBarSelectionIndicatorPresenter.cs:359]:
```suggestion
			return element?.TransformToVisual(Owner).TransformPoint(default);
```

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.Toolkit.UI/Behaviors/TabBarSelectorBehaviorState.Uwp.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ private void OnScrolViewerViewChanged(object? sender, ScrollViewerViewChangedEve
 			// -------------
 			// 1   2   3   4    
 			var offset =
-#if WINDOWS_UWP
+#if WINDOWS || WINDOWS_UWP
```
Comment (Youssef1313) [inline src/Uno.Toolkit.UI/Behaviors/TabBarSelectorBehaviorState.Uwp.cs:80]: File name ending with `.Uwp.cs` looks confusing, unless I'm missing something.

Comment (kazo0) [reply src/Uno.Toolkit.UI/Behaviors/TabBarSelectorBehaviorState.Uwp.cs:80]: yeah I'll change it to .Windows.cs

**Pull request merged**